perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -411,7 +413,7 @@ def Prepare(benchmark_spec):
   logging.info('Umount scratch disk on %s at %s', vm, mount_point)
   vm.RemoteCommand('sudo umount %s' % mount_point)
 
-  if FLAGS.device_fill_size is not '0':
```

Comment:
If against_device is true, and device_fill_size is 0, we still want to skip the device fill part.
The logic should be:
if device_fill_size is not 0:
  if device_fill_size.present or against_device:
    fill_device